picard/util/imagelist.py

```diff
@@ -18,22 +18,72 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 
+from collections.abc import MutableSequence
 
-def get_image_type(image):
-    return image.types_as_string()
+from picard import config
 
 
-class ImageList(list):
+class ImageList(MutableSequence):
+    def __init__(self, iterable=()):
+        self._images = list(iterable)
+
+    def __len__(self):
+        return len(self._images)
+
+    def __getitem__(self, index):
+        return self._images[index]
+
+    def __setitem__(self, index, value):
+        self._images[index] = value
+
+    def __delitem__(self, index):
+        del self._images[index]
+
+    # Codacy reports: Parameters differ from overridden 'insert' method
+    # but it seems perfectly legit, and matches list.insert()
+    # pylint: disable=W0221
```

Comment:
Oh i missed this, thanks for the clarification.
Fixed in https://github.com/metabrainz/picard/pull/1142/commits/8a4f1e944198e198f8c8cbb626936dd5a07fec8c

Reply:
Finally it didn't work with `element` because MutableSequence is using `value` instead...
Fixed in https://github.com/metabrainz/picard/pull/1142/commits/6ab7755f74ad47f888806a157b076ddf95e74222

Source: https://hg.python.org/cpython/file/3.5/Lib/_collections_abc.py#l895

BTW, this "rule" sucks.